test/TestProjects/Unbranded-TypeSpec/src/Generated/Models/Friend.Serialization.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System.ServiceModel.Rest;
+using System.Text.Json;
+using Azure.Core;
+
+namespace UnbrandedTypeSpec.Models
+{
+    public partial class Friend : IUtf8JsonSerializable
```

Comment:
```suggestion
    public partial class Friend : IUtf8JsonWriter
```

Reply:
We should consider changing this to IModelSerializable instead of exposing this as public.  For time constraints this might be fine to punt this out a bit.